internal/internal_task_pollers.go

```diff
@@ -164,6 +176,42 @@ func (lat *localActivityTunnel) sendTask(task *localActivityTask) bool {
 	}
 }
 
+func newLocallyDispatchedActivityTunnel(stopCh <-chan struct{}) *locallyDispatchedActivityTunnel {
+	return &locallyDispatchedActivityTunnel{
+		taskCh: make(chan *locallyDispatchedActivityTask, 25),
+		stopCh: stopCh,
+	}
+}
+
+func (ldat *locallyDispatchedActivityTunnel) getTask() *locallyDispatchedActivityTask {
+	var task *locallyDispatchedActivityTask
+	select {
+	case task = <-ldat.taskCh:
+	case <-ldat.stopCh:
+		return nil
+	}
+
+	select {
+	case ready := <-task.readyCh:
```

Comment:
Will this be head of line blocking? Not sure how big an issue it is in production. Should we consider separating to two channels internally?

Reply:
Discussed, that's related to the Yichao's comment above. If all pollers are busy with long running activities the task can stuck for a while, then we rely on the retry from the server. Will put a comment to make it clear.